### PR TITLE
docker: Adapt README for dockerhub usage

### PR DESCRIPTION
**Pull request by andre-richter**
- Add `redoxos/` namespace
- Move container building to end of README as fallback.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -34,7 +32,7 @@ docker run --cap-add MKNOD --cap-add SYS_ADMIN --device /dev/fuse \
     -e LOCAL_UID="$(id -u)" -e LOCAL_GID="$(id -g)" \
     -v redox-"$(id -u)-$(id -g)"-cargo:/usr/local/cargo \
     -v redox-"$(id -u)-$(id -g)"-rustup:/usr/local/rustup \
-    -v "$(pwd):$(pwd)" -w "$(pwd)" --rm redox make fetch all
+    -v "$(pwd):$(pwd)" -w "$(pwd)" --rm redoxos/redox make fetch all
```
Comment (fengalin) [inline docker/README.md]: If this is to be pushed to dockerhub, I would use another name than `redox` for this image as it's an image of the cross compilation toolchain, not the OS.

**Comment by fengalin** [comment]
These PR are great steps ahead!
I made a few comments.

File: docker/README.md [hunk 2]
```diff
@@ -69,3 +67,10 @@ docker run --cap-add MKNOD --cap-add SYS_ADMIN --device /dev/fuse \
 docker volume rm redox-"$(id -u)-$(id -g)"-cargo \
```
Comment (fengalin) [inline docker/README.md]: I think this section should be extended to the rustup volume.

File: docker/README.md [hunk 3]
```diff
@@ -34,7 +32,7 @@ docker run --cap-add MKNOD --cap-add SYS_ADMIN --device /dev/fuse \
     -e LOCAL_UID="$(id -u)" -e LOCAL_GID="$(id -g)" \
     -v redox-"$(id -u)-$(id -g)"-cargo:/usr/local/cargo \
     -v redox-"$(id -u)-$(id -g)"-rustup:/usr/local/rustup \
-    -v "$(pwd):$(pwd)" -w "$(pwd)" --rm redox make fetch all
+    -v "$(pwd):$(pwd)" -w "$(pwd)" --rm redoxos/redox make fetch all
 ```
 #### Linux with security modules<br>
```
Comment (fengalin) [inline docker/README.md]: Would you mind removing the `<br>` I left behind? :)

File: docker/README.md (hunk 1 above)
Comment (andre-richter) [reply docker/README.md]: Hi, thanks for the feedback.

Regarding this comment, @jackpot51 already set it up that way: https://hub.docker.com/r/redoxos/

In my opinion, the naming is okay. Redox is an OS with a non-linux kernel, so the userspace parts could not boot within within a docker container anyways. Hence not so much potential for confusion here.

How would you name it? `redox-toolchain`?

File: docker/README.md (hunk 2 above)
Comment (andre-richter) [reply docker/README.md]: Not sure I get what you mean. Can you elaborate?

File: docker/README.md (hunk 3 above)
Comment (andre-richter) [reply docker/README.md]: Force pushed a new version without it.

File: docker/README.md (hunk 1 above)
Comment (fengalin) [reply docker/README.md]: Yes, you're right. I guess `redox-toolchain` would be redundant considering the namespace, maybe something like `cross-toolchain`.

File: docker/README.md (hunk 2 above)
Comment (fengalin) [reply docker/README.md]: People might also want to remove the volume `redox-"$(id -u)-$(id -g)"-rustup`.

I guess the `Clear the named volume containing the cargo cache` section could be generalized to `Clear the named volumes`. And a command added to clean the volume `redox-"$(id -u)-$(id -g)"-rustup`.

File: docker/README.md (hunk 3 above)
Comment (fengalin) [reply docker/README.md]: Thanks

File: docker/README.md (hunk 2 above)
Comment (andre-richter) [reply docker/README.md]: Ah, it was already there in the next line, the preview window here cut it out. The headline was referring to cargo only though.
Fixed it.

**Comment by fengalin** [comment]
LGTM.

@jackpot51 what do you think about this: https://github.com/redox-os/redox/pull/1046#pullrequestreview-60299102?

File: docker/README.md (hunk 1 above)
Comment (andre-richter) [reply docker/README.md]: This is nitpicking, but is it a cross toolchain after all? The docker container is x86_64 and that is also what the target is compiled for.

`redoxos/toolchain` ?

Comment (fengalin) [reply docker/README.md]: I think we can talk about a cross toolchain as it uses a Linux host to build Redox binaries.

But as you pointed out earlier, the fact that it is a docker image makes it obvious that the host is not Redox. Under this perspective, `toolchain` is probably enough for people to understand.

**Comment by jackpot51** [comment]
Looks good! We can rename it in the future

**Pull request merged**